### ISSUE TITLE
Alibaba Cloud cannot update the software after installing the client

### ISSUE THREAD

**Issue by d960124** [tailscale/tailscale#7472]
### What is the issue?

Alibaba Cloud cannot update the software after installing the client

By default, Tailscale uses CGNAT (100.64.0.0/10) network segment as the internal address allocation network segment. At present, Tailscale only allows its own interface to use this network segment. Unfortunately, Alibaba Cloud's DNS, Apt source, etc. also use this network segment This will lead to the unavailability of DNS and Apt after Alibaba Cloud server installs the client. At present, the solution can only modify the source code, delete these two DROP rules and recompile them

code:
args = [ ]string {"!","-i",r.tunname,"-s", tsaddr.CGNATRange().String（）,"-j","DROP"}
if err := r.ipt4.Append("filter","ts-input",args...);err !=nil {
return fmt.Errorf("adding #{args} in v4/filter/ts-input: #{err}")
}

阿里云安装客户端后无法更新软件
Tailscale 默认使用 CGNAT(100.64.0.0/10) 网段作为内部地址分配网段, 目前 Tailscale 仅允许自己的接口使用此网段, 不巧的是阿里云的 DNS、Apt 源等也采用此网段. 这会导致阿里云服务器安装客户端后 DNS、Apt 等不可用, 解决方案目前只能修改源码删除掉这两个 DROP 规则并重新编译.

### OS

Linux

**Comment by d960124** [comment]
有类似问题，可以尝试新增加一条优先级高的规则：
sudo iptables -t filter -I ts-input 1 -i !tailscale0 -s 100.64.0.0/10 -j ACCEPT

注意在 -i 选项前的感叹号 (!)，这表示规则将匹配所有不是 "tailscale0" 的接口。

这条命令将在 "ts-input" 链的开头插入一条新规则，确保源 IP 地址属于阿里云内网环境（100.64.0.0/10）范围的数据包在非 "tailscale0" 接口上不被 DROP。由于它位于链的开头，优先级高于后面的规则。

**Comment by DentonGentry** [comment]
有类似问题，可以尝试新增加一条优先级高的规则：
sudo iptables -t filter -I ts-input 1 -i !tailscale0 -s 100.64.0.0/10 -j ACCEPT

注意在 -i 选项前的感叹号 (!)，这表示规则将匹配所有不是 "tailscale0" 的接口。

这条命令将在 "ts-input" 链的开头插入一条新规则，确保源 IP 地址属于阿里云内网环境（100.64.0.0/10）范围的数据包在非 "tailscale0" 接口上不被 DROP。由于它位于链的开头，优先级高于后面的规则。

--------

If you have a similar problem, you can try to add a new rule with a higher priority:
sudo iptables -t filter -I ts-input 1 -i !tailscale0 -s 100.64.0.0/10 -j ACCEPT

Note the exclamation point (!) before the -i option, which means that the rule will match all interfaces that are not "tailscale0".

This command will insert a new rule at the beginning of the "ts-input" chain to ensure that packets whose source IP addresses belong to the range of the Alibaba Cloud intranet environment (100.64.0.0/10) will not be DROPed on non-"tailscale0" interfaces. Since it is at the beginning of the chain, it takes precedence over subsequent rules.

**Comment by rmrf** [comment]
This does not work.

**Comment by d960124** [comment]
Of course, if official support isn't an option, there is another approach: launching with the `--netfilter-mode=off` flag. 
However, you must be willing to accept the following side effects. Completely disabling Netfilter renders Tailscale a "crippled" tunnel, leading to these issues: inability to automatically forward traffic (the node cannot function as a Subnet Router or Exit Node because it can no longer automatically configure Linux kernel NAT forwarding rules); reduced security (the system is no longer protected by Tailscale's native inbound rules, requiring you to manually write firewall rules to allow traffic on the `tailscale0` interface and open the listening UDP port, which defaults to 41641); and degraded connectivity (in certain complex NAT environments, the loss of firewall traversal capabilities may cause P2P direct connections to fail, forcing traffic through DERP relay servers).
--------------------------------------
当然，官方不支持的话，还有另一个办法就是：--netfilter-mode=off 参数启动
如果你能接受如下副作用的话：完全关闭 Netfilter 意味着 Tailscale 变成了一个“半残”的隧道，会带来以下问题：
无法自动转发流量：该节点将无法作为 Subnet Router（子网路由网关）或 Exit Node（出口节点）。因为它不再能自动配置 Linux 内核的 NAT 转发规则。
安全性降低：系统不再受到 Tailscale 原生入站规则的保护，你需要手动编写防火墙规则去允许 tailscale0 接口的流量和开放 UDP 监听端口（默认 41641）。
连接性下降：在某些复杂的 NAT 环境下，失去防火墙穿透配合可能导致 P2P 直连失败，从而被迫走 DERP 中继服务器。
